core/src/main/java/kafka/server/share/ShareFetchUtils.java

```diff
@@ -123,11 +134,11 @@ static Map<TopicIdPartition, ShareFetchResponseData.PartitionData> processFetchR
      *
      * @return The offset for the earliest timestamp.
      */
-    static long offsetForEarliestTimestamp(TopicIdPartition topicIdPartition, ReplicaManager replicaManager) {
+    static long offsetForEarliestTimestamp(TopicIdPartition topicIdPartition, ReplicaManager replicaManager, int leaderEpoch) {
         // Isolation level is only required when reading from the latest offset hence use Option.empty() for now.
         Option<FileRecords.TimestampAndOffset> timestampAndOffset = replicaManager.fetchOffsetForTimestamp(
                 topicIdPartition.topicPartition(), ListOffsetsRequest.EARLIEST_TIMESTAMP, Option.empty(),
-                Optional.empty(), true).timestampAndOffsetOpt();
+                Optional.of(leaderEpoch), true).timestampAndOffsetOpt();
         if (timestampAndOffset.isEmpty()) {
             throw new OffsetNotAvailableException("offset for Earliest timestamp not found for topic partition: " + topicIdPartition);
```

Comment:
nit: Capitalisation of the message is a bit odd. Why not "Offset for earliest timestamp"? Similar command for the latest case too.

Reply:
Yeah correct, something which this Pr inherited. Done.